File: machine/ts/core/system.ts
import { type MachineManifest } from '../rompack/format';

export const SYSTEM_ROM_NAME = 'bmsx-bios';
export const SYSTEM_BOOT_ENTRY_PATH = 'bios/bootrom.lua';


export const SYSTEM_MACHINE_MANIFEST: MachineManifest = {
	namespace: 'bmsx',
	vdp_class: 'psx',
};

File: machine/cpp/core/system.cpp
#include "system.h"

#include "machine/model_registry.h"

namespace bmsx {
namespace {

const MachineManifest SYSTEM_MACHINE_MANIFEST = [] {
	MachineManifest manifest;
	manifest.namespaceName = "bmsx";
	manifest.vdpClass = MachineVdpClass::Psx;
	return manifest;
}();

} // namespace

const MachineManifest& defaultSystemMachineManifest() {
	return SYSTEM_MACHINE_MANIFEST;
}

const char* systemBootEntryPath() {
	return "bios/bootrom.lua";
}

} // namespace bmsx
